File: src/internalDebug/kotlin/cash/andrew/mntrailconditions/ui/DebugUiModule.java
package cash.andrew.mntrailconditions.ui;

import cash.andrew.mntrailconditions.ui.debug.DebugViewContainer;
import cash.andrew.mntrailconditions.ui.debug.SocketActivityHierarchyServer;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module
public class DebugUiModule {
  @Provides
  @Singleton
  ViewContainer provideViewContainer(DebugViewContainer debugViewContainer) {
    return debugViewContainer;
  }

  @Provides
  @Singleton
  ActivityHierarchyServer provideActivityHierarchyServer() {
    return new SocketActivityHierarchyServer();
  }
}

File: src/internalDebug/kotlin/cash/andrew/mntrailconditions/ui/DebugUiModule.kt
package cash.andrew.mntrailconditions.ui

import cash.andrew.mntrailconditions.ui.debug.DebugViewContainer
import cash.andrew.mntrailconditions.ui.debug.SocketActivityHierarchyServer
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object DebugUiModule {
    @JvmStatic
    @Provides
    @Singleton
    fun provideViewContainer(debugViewContainer: DebugViewContainer): ViewContainer = debugViewContainer

    @JvmStatic
    @Provides
    @Singleton
    fun provideActivityHierarchyServer(): ActivityHierarchyServer = SocketActivityHierarchyServer()
}
